Tests/StreamChatUITests/Utils/DefaultMarkdownFormatter_Tests.swift

```diff
@@ -0,0 +1,252 @@
+            - struct
+            - enum
+            - actor
+            
+            For more information you can visit [this link](https://docs.swift.org/swift-book/).
+            """
+        
+        let expectedHeading1AttributedSubstring = "Swift"
+        let expectedStrikethroughAttributedSubstring = "programming"
+        let expectedBoldAttributedSubstring = "app development"
+        let expectedCodeAttributedSubstring = "let property: Double = 10.0"
+        let expectedLinkAttributedSubstring = "this link"
+        let expectedLinkURL = "https://docs.swift.org/swift-book/"
+        let expectedUnorderedListedSubstrings = ["・\tclass", "・\tstruct", "・\tenum", "・\tactor"]
+
+        // WHEN
+        let attributedString = sut.format(from: stringWithMarkdown)
+
+        // THEN
+        attributedString.enumerateAttributes(in: NSRange(
+            location: 0,
+            length: attributedString.length
+        )) { attributes, range, _ in
+            let fontAttribute = attributes[.font] as? UIFont
+            
+            if let headerAttribute = fontAttribute,
+               headerAttribute.fontDescriptor.pointSize == UIFont.preferredFont(forTextStyle: .title1).pointSize {
+                XCTAssertEqual(expectedHeading1AttributedSubstring, attributedString.attributedSubstring(from: range).string)
+            } else if let strikethroughAttribute = attributes[.strikethroughStyle] as? NSNumber,
+                      strikethroughAttribute == 1 {
+                XCTAssertEqual(expectedStrikethroughAttributedSubstring, attributedString.attributedSubstring(from: range).string)
+            } else if let boldAttribute = fontAttribute,
+                      boldAttribute.fontDescriptor.symbolicTraits.contains(.traitBold) {
+                XCTAssertEqual(expectedBoldAttributedSubstring, attributedString.attributedSubstring(from: range).string)
+            } else if let fontAttribute = fontAttribute,
+                      let fontNameAttribute = fontAttribute.fontDescriptor.fontAttributes[.name] as? String,
+                      fontNameAttribute == DefaultMarkdownFormatter.Attributes.Code.fontName {
+                XCTAssertEqual(expectedCodeAttributedSubstring, attributedString.attributedSubstring(from: range).string)
+            } else if let linkAttribute = attributes[.link] as? NSURL,
+                      let url = linkAttribute.absoluteString {
+                XCTAssertEqual(expectedLinkAttributedSubstring, attributedString.attributedSubstring(from: range).string)
+                XCTAssertEqual(expectedLinkURL, url)
+            } else if let paragraphStyleAttribute = attributes[.paragraphStyle] as? NSParagraphStyle,
+                      paragraphStyleAttribute.headIndent > 0 {
+                XCTAssertEqual(
+                    true,
+                    expectedUnorderedListedSubstrings.contains(attributedString.attributedSubstring(from: range).string)
+                )
+            }
+        }
```

Comment:
These tests are a bit too complex 🤔  One easy way to simplify this and to make it more robust is to actually use a snapshot test here. We can use a snapshot test in AttributedStrings as well 🙂 We just need to introduce a wrapper in AssertSnapshot.swift like this:
```swift
func AssertSnapshot(
    attributedString: NSAttributedString,
    record: Bool = false,
    line: UInt = #line,
    file: StaticString = #filePath,
    function: String = #function
) {
    assertSnapshot(
        matching: attributedString,
        as: .description,
        record: record,
        file: file,
        line: line
    )
}
```

Reply:
I added the test_format_whenStringContainsManyMarkdowns_thenAttributedStringIncludesAllAttibutes() test to have some verification that the returned NSAttributedString  was returning proper formatting and unfortunately that’s Apple way to test that :sweat_smile:. Ugly but is still valid. The idea of testing with a Snapshot is really good I’m gonna try that one.

Reply:
Testing with snapshot should work 👍 But we are actually testing a third-party framework, we shouldn't be testing third-party dependencies. Either way, since with snapshot testing this should be easy, we can add the test to make sure if we change the dependency, it still works 👍